src/smpp_socket.erl

```diff
@@ -955,3 +976,48 @@ pp(submit_sm_resp, _) -> record_info(fields, submit_sm_resp);
 pp(unbind, _) -> record_info(fields, unbind);
 pp(unbind_resp, _) -> record_info(fields, unbind_resp);
 pp(_, _) -> no.
+
+%%%-------------------------------------------------------------------
+%%% Overload protection
+%%%-------------------------------------------------------------------
+-spec init_rate_limit(state()) -> state().
+init_rate_limit(#{req_per_sec := undefined} = State) ->
+    State;
+init_rate_limit(#{req_per_sec := Rps} = State) ->
+    % keep one slot for eventual enquire_link request
```

Comment:
Let's not do that. Instead, on `enquire_link` timeout during the `rate_limit_cooldown`, I suggest to _insert it to gen_statem_ as a new event so it gets processed as _the first event_ after the state changes to `bound` on _state_timeout_.

Reply:
done